src/Avalonia.Controls/Primitives/SelectingItemsControl.cs

```diff
@@ -1122,6 +1333,54 @@ public Optional<int> SelectedIndex
                     _selectedIndex = default;
                 }
             }
+
+            public Optional<object?> SelectedValue
+            {
+                get => _selectedValue;
+                set
+                {
+                    _selectedValue = value;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Helper class for evaluating a binding from an Item and IBinding instance
+        /// </summary>
+        private class BindingHelper : StyledElement
```

Comment:
`BindingHelper` class here was adapted from what AutoCompleteBox does, but it requires at least a StyledElement for the DataContext property and at the very least an AvaloniaObject for the property system. Which this has the downside of rigging up a bunch of "backend" stuff when setting the data context and evaluating the binding, which may become a perf issue when scanning a large items source.

Out of curiosity, would it be (at least theoretically) possible to add an extension to evaluate a binding without needing to go through the property system? Something like:

```diff
public class BindingOperations
{
+    public object EvaluateOneTime(IBinding binding, object dataContext)
}
```
Allowing you to specify the binding and the data context and could produce a one time value without the entire property system being invoked. 
There's also other use cases for this type of thing, especially since we're trying to avoid string paths like WPF/UWP which require reflection, and we could get rid of these helper classes in AutoCompleteBox and here. 

Just a question/thought. Not sure if this is feasible though, I don't know enough about the binding internals.

Reply:
I believe EvaluateOneTime can be an internal method at the very least. And used in both controls.
About internal implementation details @grokys might have an idea, as it was planned to refactor bindings system in the future. It also can be made public at some point.